test/framework/autoscaler_helpers.go

```diff
@@ -379,7 +507,7 @@ func getAuthenticationTokenForAutoscaler(ctx context.Context, managementClusterP
 			{
 				Verbs:     []string{"get", "list"},
 				APIGroups: []string{"infrastructure.cluster.x-k8s.io"},
-				Resources: []string{infraMachineTemplateKind},
+				Resources: []string{infraMachineTemplateKind, infraMachinePoolTemplateKind, "dockermachinepools"},
```

Comment:
The idea here was to not hard-code this to CAPD so this could also be used by other infra providers. Let's please also add a infraMachinePoolKind parameter all the way through